frontend/src/components/Graph/yAxisConfig.ts

```diff
@@ -48,8 +43,33 @@ export const getYAxisFormattedValue = (
 
 export const getToolTipValue = (value: string, format: string): string => {
 	try {
+		let decimalPrecision: number | undefined;
+		const decimalSplitted = value.split('.');
+		if (decimalSplitted.length === 1) {
+			decimalPrecision = 0;
+		} else {
+			const decimalDigits = decimalSplitted[1].split('');
+			decimalPrecision = decimalDigits.length;
+			let nonZeroCtr = 0;
+			for (let idx = 0; idx < decimalDigits.length; idx += 1) {
+				if (decimalDigits[idx] !== '0') {
+					nonZeroCtr += 1;
+					if (nonZeroCtr >= 2) {
+						decimalPrecision = idx + 1;
+					}
+				} else if (nonZeroCtr) {
+					decimalPrecision = idx;
+					break;
+				}
+			}
+		}
```

Comment:
We can use less conditionals and better variable names to shorten and simplify the code path:
```
		let decimalPrecision: number;
		const decimalSplittedValues = value.split('.');
		
		if (decimalSplittedValues.length !== 1) {
			const decimalDigits = decimalSplittedValues[1].split('');
			decimalPrecision = decimalDigits.length;
			let nonZeroCtr = 0;

			for (let index = 0; index < decimalDigits.length; index += 1) {
				if (decimalDigits[index] !== '0') {
					nonZeroCtr += 1;
					if (nonZeroCtr >= 2) {
						decimalPrecision = index + 1;
					}
				} 

				if (nonZeroCtr) {
					decimalPrecision = index;
					break;
				}
			}		
		}
		
		decimalPrecision = 0;
```